artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/ssl/DefaultSSLContextFactory.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 The Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.remoting.impl.ssl;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import javax.net.ssl.SSLContext;
+import org.apache.activemq.artemis.core.remoting.impl.netty.TransportConstants;
+import org.apache.activemq.artemis.utils.ConfigurationHelper;
+
+/**
+ *  Default SSLContextFactory providing a cache using the SSL_CONTEXT_PROP_NAME value as key.
+ */
+public class DefaultSSLContextFactory extends BasicSSLContextFactory {
+
+   private static final Map<String, SSLContext> SSL_CONTEXTS = Collections.synchronizedMap(new HashMap<>());
+
+   @Override
+   public void clearSSLContexts() {
+      SSL_CONTEXTS.clear();
+   }
+
+   @Override
+   public SSLContext getSSLContext(Map<String, Object> configuration,
+           String keystoreProvider, String keystorePath, String keystorePassword,
+           String truststoreProvider, String truststorePath, String truststorePassword,
+           String crlPath, String trustManagerFactoryPlugin, boolean trustAll) throws Exception {
+      String sslContextName = getSSLContextName(configuration);
+      if (!SSL_CONTEXTS.containsKey(sslContextName)) {
+         SSL_CONTEXTS.put(sslContextName, createSSLContext(configuration,
+              keystoreProvider, keystorePath, keystorePassword,
+              truststoreProvider, truststorePath, truststorePassword,
+              crlPath, trustManagerFactoryPlugin, trustAll));
+      }
+      return SSL_CONTEXTS.get(sslContextName);
+   }
+
+   protected String getSSLContextName(Map<String, Object> configuration) {
```

Comment:
Instead of passing the `configuration` map and then potentially extracting the keystore and truststore path from it why not simply pass through the `keystorePath` and `truststorePath` which were passed in to `getSSLContext` (i.e. the caller)?

Reply:
yes sorry this is a leftover